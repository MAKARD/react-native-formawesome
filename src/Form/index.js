// @flow
export * from "./Form";
export * from "./FormProps";
