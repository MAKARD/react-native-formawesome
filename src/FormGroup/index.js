// @flow
export * from "./FormGroup";
export * from "./FormGroupProps";
