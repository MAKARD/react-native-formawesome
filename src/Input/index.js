export * from "./Input";
export * from "./InputProps";
