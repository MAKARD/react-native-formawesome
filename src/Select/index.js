// @flow
export * from "./Select";
export * from "./SelectProps";
