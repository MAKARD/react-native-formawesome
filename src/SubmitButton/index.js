export * from "./SubmitButton";
export * from "./SubmitButtonProps";
