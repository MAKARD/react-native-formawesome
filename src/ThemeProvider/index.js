// @flow
export * from "./ThemeProvider";
export * from "./ThemeProviderProps";
