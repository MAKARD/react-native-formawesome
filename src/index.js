// @flow
export * from "./Form"
export * from "./Input";
export * from "./Select";
export * from "./FormGroup";
export * from "./SubmitButton";
