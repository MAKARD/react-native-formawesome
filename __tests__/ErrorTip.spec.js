/* eslint-disable no-undef */
import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";
import { ModelValidator } from "react-formawesome-core";

import { Form, FormGroup, ErrorTip } from "../src";

class Model {
    name = undefined;
}

validator = new ModelValidator(Model);

describe("ErrorTip", () => {

    test("Should render correctly", () => {
        const tree = renderer.create(
            <Form onSubmit={() => undefined} validator={validator}>
                <FormGroup attribute="name">
                    <ErrorTip />
                </FormGroup>
            </Form>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test("Should render error text if error exist", () => {
        validator.addErrors([{
            attribute: "name",
            details: "test"
        }]);

        const tree = renderer.create(
            <Form onSubmit={() => undefined} validator={validator}>
                <FormGroup attribute="name">
                    <ErrorTip />
                </FormGroup>
            </Form>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

});
