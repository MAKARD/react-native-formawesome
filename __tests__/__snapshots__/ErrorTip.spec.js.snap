// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTip Should render correctly 1`] = `
<View>
  <View />
</View>
`;

exports[`ErrorTip Should render error text if error exist 1`] = `
<View>
  <View>
    <Text>
      test
    </Text>
  </View>
</View>
`;
